WordPress/Classes/Services/PlanService.swift

```diff
@@ -31,92 +37,87 @@ open class PlanService: LocalCoreDataService {
         let remote = PlanServiceRemote(wordPressComRestApi: api)
         remote.getWpcomPlans({ plans in
 
-            self.mergeRemoteWpcomPlans(plans.plans, remoteGroups: plans.groups, remoteFeatures: plans.features, onComplete: {
-                success()
-            })
+            self.mergeRemoteWpcomPlans(plans.plans, remoteGroups: plans.groups, remoteFeatures: plans.features, onComplete: success)
 
         }, failure: failure)
     }
 
-    func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
+    private func mergeRemoteWpcomPlans(_ remotePlans: [RemoteWpcomPlan],
                                remoteGroups: [RemotePlanGroup],
                                remoteFeatures: [RemotePlanFeature],
                                onComplete: @escaping () -> Void ) {
-
-        mergeRemoteWpcomPlans(remotePlans)
-        mergeRemotePlanGroups(remoteGroups)
-        mergeRemotePlanFeatures(remoteFeatures)
-
-        ContextManager.sharedInstance().save(managedObjectContext) {
-            onComplete()
-        }
+        coreDataStack.performAndSave({ context in
+            self.mergeRemoteWpcomPlans(remotePlans, in: context)
+            self.mergeRemotePlanGroups(remoteGroups, in: context)
+            self.mergeRemotePlanFeatures(remoteFeatures, in: context)
+        }, completion: onComplete)
     }
 
 
-    func allPlans() -> [Plan] {
+    func allPlans(in context: NSManagedObjectContext) -> [Plan] {
         let fetchRequest = NSFetchRequest<NSFetchRequestResult>(entityName: "Plan")
         fetchRequest.sortDescriptors = [NSSortDescriptor(key: "order", ascending: true)]
         do {
-            return try managedObjectContext.fetch(fetchRequest) as! [Plan]
+            return try context.fetch(fetchRequest) as! [Plan]
         } catch let error as NSError {
             DDLogError("Error fetching Plans: \(error.localizedDescription)")
             return [Plan]()
         }
```

Comment:
Out of scope of this PR, which focuses on changing the Core Data interaction, but I think it would be nice to stop logging an error and return an empty array of `Plan`s.  We could bubble up the `try` error, either by removing the `do catch` or by keeping it and re-throwing the captured error.